test/test_filesystem.cpp

```diff
@@ -18,60 +18,79 @@
 #include "c_utilities/filesystem.h"
 
 static char cwd[1024];
+#ifdef WIN32
+const char delimiter = '\\';
+#else
+const char delimiter = '/';
+#endif
 
 TEST(test_filesystem, utilities_exists) {
   utilities_get_cwd(cwd, 1024);
```

Comment:
The return value of the function call should be checked.

I would propose to even make that a "requirement" (using `__attribute__((warn_unused_result))` / `_Check_return_` on the function declaration).

Same below as well as for other functions with a return value.